### PR TITLE
Move to shared SqlClientMetaDataCollectionNames.cs

### PR DESCRIPTION
**Pull request by lcheunglci**
Relates to #1261 . Move the netcore version of SqlClientMetaDataCollectionNames.cs to shared src and update references in the netfx csproj.  The file is nearly identical between the 2 except for the slashes uses for the include file path.  The naming rule violation rule does show up for all the static variables to change them to s_camelCase but I chose not to apply them because they are public and are referenced in 28 places, but they are all from the 2 unit tests `SqlClientMetaDataCollectionNamesTest.cs` and `ConnectionSchemaTest.cs`.  If those should be updated. I can make the changes to fix the coding style violations and update the tests.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Data.SqlClient
+{
+
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs:7]:
```suggestion
```

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        public static readonly string Users = "Users";
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlClientMetaDataCollectionNames.xml' path='docs/members[@name="SqlClientMetaDataCollectionNames"]/ViewColumns/*' />
+        public static readonly string ViewColumns = "ViewColumns";
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlClientMetaDataCollectionNames.xml' path='docs/members[@name="SqlClientMetaDataCollectionNames"]/Views/*' />
+        public static readonly string Views = "Views";
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlClientMetaDataCollectionNames.xml' path='docs/members[@name="SqlClientMetaDataCollectionNames"]/AllColumns/*' />
+        public static readonly string AllColumns = "AllColumns";  // supported starting from SQL Server 2008
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs:48]: Minimum SQL Server supported is SQL Server 2012, comment no longer applicable.
```suggestion
        public static readonly string AllColumns = "AllColumns";
```

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlClientMetaDataCollectionNames.xml' path='docs/members[@name="SqlClientMetaDataCollectionNames"]/ColumnSetColumns/*' />
+        public static readonly string ColumnSetColumns = "ColumnSetColumns";  // supported starting from SQL Server 2008
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientMetaDataCollectionNames.cs:51]:
```suggestion
        public static readonly string ColumnSetColumns = "ColumnSetColumns";
```

**Review by cheenamalhotra — APPROVED** [review]
Public APIs should not be modified, so API names can be left as is.
LGTM, just some minor touch-ups needed.

**Review by JRahnama — APPROVED** [review]

**Pull request merged**